tensorrt_llm/_torch/speculative/interface.py

```diff
@@ -42,7 +46,7 @@ def extend_ctx(self):
         chunked context requests at the kernel level. Required for
         any spec dec mode that uses the SpecExecutor.
         """
-        return self.is_eagle3()
+        return self.is_eagle3() or self.is_ngram()
```

Comment:
Looks like this needs a rebase, there's some extra logic here on the latest main

Reply:
Rebased.